src/NuGet.Core/NuGet.Build.Tasks/RestoreTaskEx.cs

```diff
@@ -19,13 +19,13 @@ namespace NuGet.Build.Tasks
     /// </summary>
     public sealed class RestoreTaskEx : Task, ICancelableTask, IDisposable
     {
+        public readonly CancellationTokenSource CancellationTokenSource = new CancellationTokenSource();
```

Comment:
Can this be `internal` instead of `public`?  If the only need is from tests, consider `InternalsVisibleTo` over `public`.  If it must be `public` for other reasons, make it a read-only property.

Reply:
Sure